src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -492,196 +543,248 @@ protected void Accepting(object message)
                         .ContinueWith(x =>
                         {
                             return new ManagementCommandAck(x.Result.All(y => y));
-                        },
-                            TaskContinuationOptions.ExecuteSynchronously)
+                        }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default)
                         .PipeTo(sender);
-                })
-                .With<Quarantine>(quarantine =>
-                {
+                });
+
+            Receive<Quarantine>(quarantine =>
+            {
                     //Stop writers
-                    if (endpoints.WritableEndpointWithPolicyFor(quarantine.RemoteAddress) is Pass)
-                    {
-                        var pass = (Pass) endpoints.WritableEndpointWithPolicyFor(quarantine.RemoteAddress);
+                    var policy =
+                    Tuple.Create(_endpoints.WritableEndpointWithPolicyFor(quarantine.RemoteAddress), quarantine.Uid);
+                if (policy.Item1 is Pass && policy.Item2 == null)
```

Comment:
If the current write policy for this endpoint is `Pass` (writes are allowed to flow through to the transport) and we do not have a UID available for the `Address` we are going to `Quarantine`, stop the endpoint actor (a `ReliableDeliverySupervisor`) and mark the endpoint as failed.

We can't apply a `Quarantine` policy to an endpoint without know its UID, so we will `Gate` the connection instead.